person.rb

```diff
@@ -0,0 +1,47 @@
+class Person
+  def initialize(id, age, parent_permission: true, name: 'Unknown')
+    @id = id
+    @id = id
+    @age = age
+    @parent_permission = parent_permission
+    @name = name
+  end
+
+  def add_id(id)
+    @id = id
+  end
+
+  # Get name method
+  def add_name(name)
+    @name = name
+  end
+
+  # Get age method
+  def add_age(age)
+    @age = age
+  end
```

Comment:
- [ ] The comment above the method on line 14 and 19 are kind of misleading. The `add_name` and `add_age` should be a setter method and not a getter. Then the other methods which should be getter methods should not accept any argument since it just returns the instance variable `@name` or `@age`. Also notice the `=` sign at the end of the setter methods. 

```ruby

class Person
  def add_name=(name)  # Setter method for name
    @name = name
  end

  def add_age=(age)  # Setter method for age
    @age = age
  end

  def name  # Getter method for name
    @name
  end

  def age  # Getter method for age
    @age
  end
end
```

- `add_name=(name)`: This is a setter method for the name property. The = sign at the end of the method name is part of Ruby's syntactic sugar for setter methods. It takes a name parameter and assigns it to the instance variable @name. This method allows you to set the name property of a Person object.

- `add_age=(age)`: Similarly, this is a setter method for the age property. It takes an age parameter and assigns it to the instance variable @age. This method allows you to set the age property of a Person object.

- `name`: This is a getter method for the name property. It simply returns the value stored in the @name instance variable.

- [ ] Alternatively, you can use the short syntax for exposing a setter and getter methods like so 👇🏾 

```ruby
 # *:name and :age instance variables are the only ones that should be able to be modified that's why they are defined as attr_accessors (here both Getters and Setters are defined)
  attr_accessor :name, :age 
  # *:id should only be able to be retrieved so only the Getter is defined
  attr_reader :id 
```

- `age`: This is a getter method for the age property. It returns the value stored in the @age instance variable.